python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1573 @@
+                self.tarts.alignnet_model,
+                self.tarts.wavenet_model,
+                self.tarts.aggregatornet_model,
+            ]:
+                if model is not None:
+                    model.device_val = torch.device("cpu")
+
+                    # Update device attributes for underlying models and CNNs
+                    if hasattr(model, "alignnet"):
+                        model.alignnet.device_val = torch.device("cpu")
+                        model.alignnet.cnn = model.alignnet.cnn.cpu()
+                    if hasattr(model, "wavenet"):
+                        model.wavenet.device_val = torch.device("cpu")
+                        model.wavenet.cnn = model.wavenet.cnn.cpu()
+            self.log.debug("Moved all sub-models and CNNs to CPU")
+        else:
+            self.tarts.to("cuda")
+            self.log.debug("Moved models to CUDA")
+
+        self.log.debug("TARTS crop size: %s", self.cropSize)
+
+    def _populateStampMetadata(
+        self,
+        row: dict,
+        stamp: Optional[DonutStamp],
+        stamps: DonutStamps,
+        key_to_column: dict[str, str],
+        i: int,
+        prefix: str,
+    ) -> None:
+        """Populate row with metadata from a single stamp.
+
+        Parameters
+        ----------
+        row : dict
+            The row dictionary to populate.
+        stamp : DonutStamp or None
+            The individual stamp (can be None if no stamp at this index).
+        stamps : DonutStamps
+            The stamps collection containing metadata.
+        key_to_column : dict[str, str]
+            Mapping from metadata keys to column name suffixes.
+        i : int
+            Index into the metadata arrays.
+        prefix : str
+            Column prefix ("intra" or "extra").
+        """
+        for key, col_suffix in key_to_column.items():
+            if stamp is not None and stamps.metadata.exists(key):
+                val = stamps.metadata.getArray(key)[i]
+                # Add units for centroid columns (pixels)
+                if key in ["CENT_X", "CENT_Y"]:
+                    row[f"{prefix}_{col_suffix}"] = val * u.pixel
+                else:
+                    row[f"{prefix}_{col_suffix}"] = val
+                # field_x/y are aliases for fx/fy
+                if key == "FX":
+                    row[f"{prefix}_field_x"] = val * u.deg
+                elif key == "FY":
+                    row[f"{prefix}_field_y"] = val * u.deg
+            else:
+                # Add appropriate units for NaN values
+                if key in ["CENT_X", "CENT_Y"]:
+                    row[f"{prefix}_{col_suffix}"] = np.nan * u.pixel
+                else:
+                    row[f"{prefix}_{col_suffix}"] = np.nan
+                if key == "FX":
+                    row[f"{prefix}_field_x"] = np.nan * u.deg
+                elif key == "FY":
+                    row[f"{prefix}_field_y"] = np.nan * u.deg
+
+    def validate(self) -> None:
```

Comment:
I don't think this is called anywhere? Do you want to call it when instantiating?